pkg/predicate/predicate.go

```diff
@@ -51,3 +58,37 @@ func (GenerationChangedPredicate) Update(e event.UpdateEvent) bool {
 	}
 	return true
 }
+
+// Skips events that have labels matching a selector defined in watches.yaml
+func EventFilter(eventLabels map[string]string, r ResourceFilterPredicate) bool {
+
+	requirements, err := metav1.LabelSelectorAsSelector(&r.Selector)
+
+	if err != nil {
+		log.Error(nil, "Unable to retrieve requirement", err)
+		return false
```

Comment:
Hmm, if we're unable to parse the requirement should our behavior be to reconcile or not? My inclination is to reconcile in this case but curious what others think